packages/client/components/PokerEstimateHeaderCardParabol.tsx

```diff
@@ -87,17 +98,37 @@ const Content = styled('div')({
   paddingRight: 4
 })
 
+const CardTitle = styled('h1')({
+  fontSize: 16,
+  lineHeight: '24px',
+  margin: '0 0 8px'
+})
+
+const CardTitleWrapper = styled('div')({
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'flex-start',
+  width: '100%'
+})
+
+const CardDescription = styled('div')({
+  color: PALETTE.SLATE_700,
+  fontWeight: 'normal',
+  lineHeight: '20px',
+  fontSize: 14,
+  margin: 0,
+  height: '100%'
+})
+
 interface Props {
   stage: PokerEstimateHeaderCardParabol_stage
 }
 
-type Story = Required<NonNullable<NonNullable<PokerEstimateHeaderCardParabol_stage>['story']>>
-
 const PokerEstimateHeaderCardParabol = (props: Props) => {
   const {stage} = props
-  const {story} = stage
-  const {content, id: taskId, teamId} = story as Story
-  const integration = story!.integration
+  const story = stage.story as Extract<typeof stage['story'], {__typename: 'Task'}> | null
```

Comment:
@nickoferrall i know we weren't happy with how we were doing this, so i found a neat little hack. if you request the `__typename` field explicitly in the fragment, then relay gives you a discriminated union. you can then `Extract` the type you want, which means nullability for the story fields will be preserved.

Reply:
Nice! That's good to know